src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -597,7 +594,8 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
                 }
 
                 // Allow nullable parameters that don't have a default value
-                if (nullability.ReadState == NullabilityState.Nullable && !parameter.HasDefaultValue)
+                var nullability = NullabilityContext.Create(parameter);
+                if (nullability.ReadState != NullabilityState.NotNull && !parameter.HasDefaultValue)
```

Comment:
Does this work?

```suggestion
                if (isOptional)
```

Reply:
This doesn't work because we want to check if there is no default value provided (which is the opposite of the isOptional check). We can do something like:

```
if (isOptional && !parameter.HasDefaultValue) {
```

But that's a little confusing to read so that's why I kept it this way.

Reply:
Deserves a comment?